src/message_private.h

```diff
@@ -74,7 +74,12 @@ typedef struct {
 //
 
 typedef struct {
-    sys_mutex_t          lock;
+    sys_mutex_t          lock,
+                         producer_lock,                   // These locks prevent either side from activating
```

Comment:
These should have more descriptive, less misleading names.  {producer|consumer}_activation_lock